### PR TITLE
Use real SPE data in fax

### PR DESCRIPTION
**Pull request by ershockley**
Add spe_distribution.csv which contains pdf for sample channel. Modify simulation.py to pull from this distribution for the gains rather than truncated gaussian

### CODE REVIEW

File: pax/simulation.py [hunk 1]
```diff
@@ -801,3 +801,29 @@ def truncated_gauss_rvs(my_mean, my_std, left_boundary, right_boundary, n_rvs):
     See http://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.stats.truncnorm.html
     """
     return _truncated_gauss(my_mean, my_std, left_boundary, right_boundary).rvs(n_rvs) * my_std + my_mean
+
+def load_spe_pdf(file):
```
Comment (tunnell) [inline pax/simulation.py]: __[flake8]__
* [E302] expected 2 blank lines, found 1

File: pax/simulation.py [hunk 2]
```diff
@@ -801,3 +801,29 @@ def truncated_gauss_rvs(my_mean, my_std, left_boundary, right_boundary, n_rvs):
...
+    """Extract the spe pdf from a csv file in form of bin, prob. Returns two arrays."""
+    charge = []
+    p = []
+    with open(file, "r") as f:
+        for line in f.readlines():
+            q = float(line.split(",")[0])
+            prob = float(line.split(",")[1])
+            # don't allow negative charge, which can occur from noise subtraction
+            prob = prob if c >= 0 else 0
```
Comment (tunnell) [inline pax/simulation.py]: __[flake8]__
* [F821] undefined name 'c'

File: pax/simulation.py [hunk 3]
```diff
@@ -801,3 +801,29 @@ def truncated_gauss_rvs(my_mean, my_std, left_boundary, right_boundary, n_rvs):
...
+            charge.append(q)
+            p.append(prob)
+
+    return np.array(charge), np.array(p)
+
+def generate_gains(bins, pdf, N):
```
Comment (tunnell) [inline pax/simulation.py]: __[flake8]__
* [E302] expected 2 blank lines, found 1

File: pax/simulation.py [hunk 4]
```diff
@@ -801,3 +801,29 @@ def truncated_gauss_rvs(my_mean, my_std, left_boundary, right_boundary, n_rvs):
...
+    """ Take the charge bins, pdf of spe distribution and generate sample of N gains"""
+    cdfs = np.cumsum(pdf)/np.sum(pdf)
+    Interpolator = interp1d(cdfs, bins)
+    random_cdfs = np.random.uniform(0, 1, N)
+    gains = []
+    for cdf in random_cdfs:
+        gains.append( Interpolator(cdf))
```
Comment (tunnell) [inline pax/simulation.py]: __[flake8]__
* [E201] whitespace after '('

File: pax/simulation.py [hunk 5]
```diff
@@ -801,3 +801,29 @@ def truncated_gauss_rvs(my_mean, my_std, left_boundary, right_boundary, n_rvs):
...
+    return gains
+
```
Comment (tunnell) [inline pax/simulation.py]: __[flake8]__
* [W391] blank line at end of file

File: pax/simulation.py [hunk 6]
```diff
@@ -104,6 +104,35 @@ def __init__(self, config_to_init):
...
+                    charge.append(q)
+                    p.append(prob)
+
+            # Create a converter from uniform random numbers to SPE gains
+            # note not really pdf, does not necessarily sum to 1 due to forcing probability of negative charge to be 0
+            spe_bins, spe_pdf = np.array(charge), np.array(p)
+            # change spe_bins to have mean equal to 1
+            mean_spe = (spe_bins * spe_pdf).sum() / spe_pdf.sum()
+            spe_bins = spe_bins/mean_spe
+            #convert pdf -> cdf
```
Comment (tunnell) [inline pax/simulation.py]: __[flake8]__
* [E265] block comment should start with '# '

**Comment by ershockley** [comment]
I just made a change that normalizes the spe distribution to have mean equal to one (after loading the file). Once midway is back up I can modify it in the file itself. Here's the distribution now

spe_dist_mean

**Pull request merged**